src/createrepo_c.c

```diff
@@ -880,48 +882,53 @@ main(int argc, char **argv)
+            }
 
-        //compress new module metadata string to a file in temporary .repodata
-        gchar *modules_metadata_path = g_strconcat(tmp_out_repo, "modules.yaml", compression_suffix, NULL);
-        CR_FILE *modules_file = NULL;
-        modules_file = cr_open(modules_metadata_path, CR_CW_MODE_WRITE, compression, &tmp_err);
-        if (modules_file == NULL) {
-            g_critical("%s: Cannot open source file %s: %s", __func__, modules_metadata_path,
-                       (tmp_err ? tmp_err->message : "Unknown error"));
-            g_clear_error(&tmp_err);
+            //compress new module metadata string to a file in temporary .repodata
+            gchar *modules_metadata_path = g_strconcat(tmp_out_repo, "modules.yaml", compression_suffix, NULL);
+            CR_FILE *modules_file = NULL;
+            modules_file = cr_open(modules_metadata_path, CR_CW_MODE_WRITE, compression, &tmp_err);
+            if (modules_file == NULL) {
+                g_critical("%s: Cannot open source file %s: %s", __func__, modules_metadata_path,
+                           (tmp_err ? tmp_err->message : "Unknown error"));
+                g_clear_error(&tmp_err);
+                free(moduleindex_str);
+                free(modules_metadata_path);
+                exit(EXIT_FAILURE);
+            }
+            cr_puts(modules_file, moduleindex_str, &tmp_err);
             free(moduleindex_str);
-            free(modules_metadata_path);
-            exit(EXIT_FAILURE);
-        }
-        cr_puts(modules_file, moduleindex_str, &tmp_err);
-        free(moduleindex_str);
-        cr_close(modules_file, &tmp_err);
-        if (tmp_err) {
-            g_critical("%s: Error while closing: : %s", __func__, tmp_err->message);
-            g_clear_error(&tmp_err);
-            free(modules_metadata_path);
-            exit(EXIT_FAILURE);
+            cr_close(modules_file, &tmp_err);
+            if (tmp_err) {
+                g_critical("%s: Error while closing: : %s", __func__, tmp_err->message);
+                g_clear_error(&tmp_err);
+                free(modules_metadata_path);
+                exit(EXIT_FAILURE);
+            }
+
+            //create additional metadatum for new module metadata file
+            cr_Metadatum *new_modules_metadatum = g_malloc0(sizeof(cr_Metadatum));
+            new_modules_metadatum->name = modules_metadata_path;
+            new_modules_metadatum->type = g_strdup("modules");
+            additional_metadata = g_slist_prepend(additional_metadata, new_modules_metadatum);
         }
 
-        //create additional metadatum for new module metadata file
-        cr_Metadatum *new_modules_metadatum = g_malloc0(sizeof(cr_Metadatum));
-        new_modules_metadatum->name = modules_metadata_path;
-        new_modules_metadatum->type = g_strdup("modules");
-        additional_metadata = g_slist_prepend(additional_metadata, new_modules_metadatum);
+        g_clear_pointer(&merger, g_object_unref);
```

Comment:
Why did you move clearing the merger to the end? This way, the code can exit without clearing it first in case of failure. Or am I missing something?

Reply:
O right, it must be cleared in case merger_is_empty. But it still holds that the pointer is not cleared in case it exits earlier during the if.

Reply:
Fixed :+1: